src/index.ts

```diff
@@ -91,9 +91,24 @@ declare module "http" {
   }
 }
 
+type RequestType =
+  | { headers: IncomingHttpHeaders }
```

Comment:
Why not use the actual types instead of creating ones?

Reply:
(I guess: because that would make this file less generic and more tied to Node/Next?)

Reply:
The actual types (`NextRequest`, `NextResponse`) aren't there in older Next.js versions (`iron-session` requires >=10).

Reply:
But yes, please make it more generic, e.g. making it compatible with the Web API / Service Worker API, which in turn makes it compatible for Deno / Cloudflare Workers.

Reply:
We did it here: (we did not assume that in an update to this file: https://github.com/vvo/iron-session/pull/510/commits/b9fc40839313ebe282b0dd1f7f621a403bfe07be, more to come!)